lib/ash_authentication/strategies/api_key/plug.ex

```diff
@@ -0,0 +1,185 @@
+defmodule AshAuthentication.Strategy.ApiKey.Plug do
+  @moduledoc """
+  Plug for authenticating using API keys.
+
+  This plug validates API keys from either a query parameter or HTTP header.
+  """
+
+  @behaviour Plug
+
+  alias Ash.PlugHelpers
+  alias AshAuthentication.{Info, Strategy}
+  alias Plug.Conn
+
+  @type source_type :: :header | :query_param | :header_or_query_param
+  @type auth_error :: :invalid_api_key | :missing_api_key | :authentication_failed
+
+  @doc """
+  Handles errors that occur during the api key authentication process.
+
+  This function determines the response format based on the `Accept` header
+  of the incoming request. If the client accepts JSON responses, it returns
+  a JSON-formatted error message. Otherwise, it returns a plain text error
+  message.
+
+    - If the `Accept` header contains "json", the response will be:
+      - Status: 403 Forbidden
+      - Content-Type: application/json
+      - Body: `{"error":"Forbidden"}`
+    - Otherwise, the response will be:
+      - Status: 403 Forbidden
+      - Content-Type: text/plain (default)
+      - Body: `Forbidden`
+  """
+  def on_error(conn, _error) do
+    if Plug.Conn.get_req_header(conn, "accept") |> Enum.any?(&String.contains?(&1, "json")) do
+      conn
+      |> Plug.Conn.put_resp_content_type("application/json")
+      |> Plug.Conn.send_resp(403, ~s({"error":"Forbidden"}))
```

Comment:
An invalid token usually is 401...

```suggestion
      |> Plug.Conn.send_resp(401, ~s({"error":"Unauthorized"}))
```